app/scripts/components/common/catalog/controls/atoms/taxonomy-atom.ts

```diff
@@ -0,0 +1,19 @@
+import { atomWithUrlValueStability } from '$utils/params-location-atom/atom-with-url-value-stability';
+
+export const taxonomyAtom = atomWithUrlValueStability<Record<string, string | string[]>, Record<string, string | string[]>>({
+  initialValue: {},
+  urlParam: 'taxonomy',
```

Comment:
I created a `taxonomy` and `search` atoms as the `useBrowseControls` did not seem to consistently sync the url param after clearing all the taxonomy filters.

For example, I would select few taxonomies, select a dataset, add it to a map. Then re-open the modal, deselect the prev selected taxonomies, select another dataset and add it to a map. Reopen the modal, the taxonomies are persisted because they pop up again in the `url`.

This might have to do with the fact that in E&A we sync query params from two places, once using jotai atoms (ie. `datasets`, `aois`) and once now using the `useBrowseControls` which uses a custom url string management lib. 

To fix the issue I came up with these two atoms and a custom hook which updates / reads from them (similar to what we already have for the rest of the E&A url params).

Let me know what do you think about this.

Reply:
Adding a few more problems that I found

- Atom values perseist through pages ex. users interact with the modal on the E&A page and navigate to the catalog page. Search and Taxonomy values that were selected in E&A page pop up. I don't think this is the expected behavior
- Something is weird when there is more than one selection happens. The interaction from the third interaction doesn't seem to work at all. ex. in the video below, I selected 1. Topics>Air Quality, 2. Gas>Co2, then unselected Gas>Co2, but the result doesn't show the result from step 1.

https://github.com/NASA-IMPACT/veda-ui/assets/4583806/7c435e46-f9c9-440f-85e5-3621db8c2f73

Reply:
Thanks for the QA and the insightful comment! All should be addressed now. 

> I am unsure if we need URLs to be manipulated when users interact through modal

I agree (I was trying to recreate the same behavior from the legacy modal). I went ahead and stopped persisting the taxonomies in the URL, but they will persist in a local state if I close and reopen the modal until i refresh the page as I think it's nice to have the immediate previous selection I've made available to me as a user. But it's an easy tweak for me to clear this state out if you think otherwise!
 
 > Atom values perseist through pages ex. users interact with the modal on the E&A page and navigate to the catalog page
> Maybe we can use the hook only for Data Catalog page?
 
Fixed. We'll use the hook in the Data Catalog page and a local state in the Modal, and pass the taxonomies down to the Catalog Content (remember the point from yesterday when we discussed how we should move the taxonomies up one level? 😃)
 
 > Something is weird when there is more than one selection happens
 
 Should be fixed